nomad/structs/structs.go

```diff
@@ -316,12 +317,20 @@ type NodeRegisterRequest struct {
 }
 
 // NodeDeregisterRequest is used for Node.Deregister endpoint
-// to deregister a node as being a schedulable entity.
+// to deregister a batch of nodes from being schedulable entities.
+// Deprecated in 0.9.3
 type NodeDeregisterRequest struct {
 	NodeID string
 	WriteRequest
 }
 
+// NodeDeregisterBatchRequest is used for Node.DeregisterBatch endpoint
+// to deregister a batch of nodes from being schedulable entities.
+type NodeDeregisterBatchRequest struct {
```

Comment:
I find our naming conventions here odd, Batch seems to appear inconsistently throughout the file: `BatchNodeUpdateDrainRequest`, `JobBatchDeregisterRequest`, and `NodeDeregisterBatchRequest` :(.  The same goes for the fsm method handler name.

Reply:
agreed. I'll rename it to avoid introducing a third pattern at least